File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate){
    int len = 0, sum = 0, n = candidate, m = candidate;
    while (n){
        n /= 10;
        len++;
    }
    for (int i = 0; i < len; i++){
        sum += pow(m%10, len);
        m /= 10;
    }
    return sum == candidate;
}

File: solutions/julia/armstrong-numbers/1/armstrong-numbers.jl
function isarmstrong(num)
    len = 0; sum = 0; m = num; n = num
    while n != 0
        n ÷= 10
        len += 1
    end
    for i in 1:len
        sum += (m % 10)^len
        m ÷= 10
    end
    sum == num
end